lib/fizzy/execute.cpp

```diff
@@ -418,13 +419,14 @@ template <typename T>
 inline T shift_right(T lhs, T rhs) noexcept
 {
     constexpr T num_bits{sizeof(T) * 8};
-    const auto k = rhs % num_bits;
+    const auto k = rhs & (num_bits - 1);
```

Comment:
This is good change (still single CPU instruction). So maybe use the same pattern in all other cases?

Reply:
Changed it in all shifts